test/catalog/index_handle_test.cpp

```diff
@@ -0,0 +1,111 @@
+#include "catalog/catalog.h"
+#include "transaction/transaction_manager.h"
+#include "type/transient_value_factory.h"
+#include "type/transient_value_peeker.h"
+#include "util/test_harness.h"
+#include "util/transaction_test_util.h"
+
+namespace terrier {
+struct IndexHandleTest : public TerrierTest {
+  void SetUp() override {
+    TerrierTest::SetUp();
+    txn_manager_ = new transaction::TransactionManager(&buffer_pool_, true, LOGGING_DISABLED);
+
+    txn_ = txn_manager_->BeginTransaction();
+    catalog_ = new catalog::Catalog(txn_manager_, txn_);
+  }
+
+  void TearDown() override {
+    txn_manager_->Commit(txn_, TestCallbacks::EmptyCallback, nullptr);
+    TerrierTest::TearDown();
+    delete catalog_;  // delete catalog first
+    delete txn_manager_;
+    delete txn_;
+  }
+
+  catalog::Catalog *catalog_;
+  storage::RecordBufferSegmentPool buffer_pool_{100, 100};
+
+  transaction::TransactionContext *txn_;
+  transaction::TransactionManager *txn_manager_;
+};
+
+// Tests that we can correctly add new index entry to the pg_index and correctly get the index entry by the index oid.
+// NOLINTNEXTLINE
```

Comment:
[Major] Probably need another test to test `DeleteEntry()`

Reply:
Added.